### ISSUE TITLE
Assignment #1, cross-entropy cost

### ISSUE THREAD

**Issue by yurket** [ferrarisf50/CS224n#1]
Hi there! My name is Yury and I'm doing cs224n programming assignment right now. I examined your code (to find an error in mine) and don't get at least 1 thing: why you don't have a 1/N term in cross-entropy cost here https://github.com/ferrarisf50/CS224n/blob/master/assignment1/q2_neural.py#L43?
I'd expect to see here something like`cost=-np.sum(np.log(yhat)*labels) / M`, where `M = len(data)`. Could you explain this, please? =) Thanks!

**Comment by ferrarisf50** [comment]
Thanks Yury. I think you're right. Cross Entropy function should be averaged. I am a beginner on NLP and not finishing Assignment 1 yet so there could be some mistakes.

**Comment by yurket** [comment]
No problem) I'm a beginner too, so could be wrong as well. Don't forget to place 1/M in the d1 variable, so that `d1= (yhat-labels) / M`, because 1/M term propagates from the cost derivative.
Meanwhile, I have another question: in this line https://github.com/ferrarisf50/CS224n/blob/master/assignment1/q2_neural.py#L54 why you have `sigmoid_grad(h)`, and not `sigmoid_grad(x1)`? I asked the same question here: https://github.com/mnvl/cs224n/issues/1

**Comment by yurket** [comment]
Sorry for bothering - just figured it out (left a comment in an original issue).
